bin/run.sh

```diff
@@ -37,6 +37,11 @@ file_contents=$(< "${input_dir}/stack.yaml")
 
 echo "system-ghc: true" >> "${input_dir}/stack.yaml"
 
+# Run our SetupTestFile which does some code injection to modify how the tests
+# will run to use our custom hspec formatter that outputs results.json automatically
+# TODO: see if we can avoid runghc and instead run a precompiled binary
```

Comment:
It would definitely be good if we could use a pre-compiled binary, as that would likely improve the speed.

Reply:
I will see what I can do about this.
In the mean time, could you please approve the Github Actions workflow to run on the latest commits so that I can see whether the tests will pass there?

Reply:
Will do!

Reply:
I think there might be an issue with the latest changes. Sorry about that. I've been switching between a Mac and Linux and the behaviour is completely different between the two even when using Docker which makes it kinda difficult to know if my changes are working or not.

I am currently debugging on my Linux machine to try and see what's going on there.

Reply:
OK should be fixed now.